Integrations/MicrosoftGraphListener/MicrosoftGraphLIstener.py

```diff
@@ -0,0 +1,985 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+''' IMPORTS '''
+import requests
+import base64
+import os
+import json
+from cryptography.hazmat.primitives.ciphers.aead import AESGCM
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' GLOBAL VARS '''
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+
+# Well known folders shortcut in MS Graph API
+# For more information: https://docs.microsoft.com/en-us/graph/api/resources/mailfolder?view=graph-rest-1.0
+WELL_KNOWN_FOLDERS = {
+    'archive': 'archive',
+    'conversation history': 'conversationhistory',
+    'deleted items': 'deleteditems',
+    'drafts': 'drafts',
+    'inbox': 'inbox',
+    'junk email': 'junkemail',
+    'outbox': 'outbox',
+    'sent items': 'sentitems',
+}
+
+EMAIL_DATA_MAPPING = {
+    'id': 'ID',
+    'createdDateTime': 'CreatedTime',
+    'lastModifiedDateTime': 'ModifiedTime',
+    'receivedDateTime': 'ReceivedTime',
+    'sentDateTime': 'SentTime',
+    'subject': 'Subject',
+    'importance': 'Importance',
+    'conversationId': 'ConversationID',
+    'isRead': 'IsRead',
+    'isDraft': 'IsDraft',
+    'internetMessageId': 'MessageID'
+}
+
+''' HELPER FUNCTIONS '''
+
+
+def epoch_seconds(d=None):
+    """
+    Return the number of seconds for given date. If no date, return current.
+
+    :type d: ``datetime``
+    :param d: Datetime
+
+    :return: Timestamp in epoch
+    :rtype: ``int``
+    """
+    if not d:
+        d = datetime.utcnow()
+    return int((d - datetime.utcfromtimestamp(0)).total_seconds())
```

Comment:
consider switching this:
https://blog.ganssle.io/articles/2019/11/utcnow.html

Reply:
switched